exporter/loadbalancingexporter/resolver_aws_cloudmap.go

```diff
@@ -0,0 +1,218 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package loadbalancingexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/service/servicediscovery/types"
+	"go.opencensus.io/tag"
+	"log"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/servicediscovery"
+
+	"go.opencensus.io/stats"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultAwsResInterval = 30 * time.Second
+	defaultAwsResTimeout  = time.Second
+)
+
+var (
+	errNoNamespace   = errors.New("no Cloud Map namespace specified to resolve the backends")
+	errNoServiceName = errors.New("no Cloud Map serviceName specified to resolve the backends")
+
+	awsResolverMutator = tag.Upsert(tag.MustNewKey("resolver"), "aws")
+
+	awsResolverSuccessTrueMutators  = []tag.Mutator{awsResolverMutator, successTrueMutator}
+	awsResolverSuccessFalseMutators = []tag.Mutator{awsResolverMutator, successFalseMutator}
+)
+
+func createDiscoveryFunction(client *servicediscovery.Client) func(params *servicediscovery.DiscoverInstancesInput) (*servicediscovery.DiscoverInstancesOutput, error) {
+	return func(params *servicediscovery.DiscoverInstancesInput) (*servicediscovery.DiscoverInstancesOutput, error) {
+		return client.DiscoverInstances(context.TODO(), params)
+	}
+}
+
+type cloudMapResolver struct {
+	logger *zap.Logger
+
+	namespaceName *string
+	serviceName   *string
+	port          *uint16
+	healthStatus  *types.HealthStatusFilter
+	resInterval   time.Duration
+	resTimeout    time.Duration
+
+	endpoints         []string
+	onChangeCallbacks []func([]string)
+
+	stopCh             chan struct{}
+	updateLock         sync.Mutex
+	shutdownWg         sync.WaitGroup
+	changeCallbackLock sync.RWMutex
+	discoveryFn        func(params *servicediscovery.DiscoverInstancesInput) (*servicediscovery.DiscoverInstancesOutput, error)
+}
+
+func newCloudMapResolver(logger *zap.Logger, namespaceName *string, serviceName *string, port *uint16, healthStatus *types.HealthStatusFilter, interval time.Duration, timeout time.Duration) (*cloudMapResolver, error) {
+	// Using the SDK's default configuration, loading additional config
+	// and credentials values from the environment variables, shared
+	// credentials, and shared configuration files
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithDefaultRegion("us-east-1"))
+	if err != nil {
+		log.Fatalf("unable to load SDK config, %v", err)
+	}
+
+	// Using the Config value, create the DynamoDB client
+	svc := servicediscovery.NewFromConfig(cfg)
+
+	if len(*namespaceName) == 0 {
```

Comment:
It might be good to do a nil check and a len check:

```
if namespaceName == nil || len(*namespaceName) == 0 {
  ...
}
```

Otherwise, you might have a runtime panic if namespaceName is nil. Same for serviceName